weasyl/moderation.py

```diff
@@ -719,52 +719,88 @@ def note_about(userid, target_user, title, message=None):
         ))
 
 
+def get_userid_from_loginname(login_name):
+    """Selects a userid by their login_name.
+
+    Args:
+        login_name (str): Login name to get ID for.
+
+    Returns:
+        An int.
+    """
+
+    db = d.connect()
+
+    user = d.meta.tables['login']
+
+    query = d.sa.select([
+        user.c.userid,
+    ]).where(user.c.login_name.op('~')(login_name))
```

Comment:
It appears to take a regular expression to match a login name, not a login name.